server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/WriteLoadConstraintDecider.java

```diff
@@ -0,0 +1,124 @@
+        var allShardWriteLoads = allocation.clusterInfo().getShardWriteLoads();
+        var shardWriteLoad = allShardWriteLoads.get(shardRouting.shardId());
+        if (shardWriteLoad == null || shardWriteLoad == 0) {
+            return Decision.single(Decision.Type.YES, NAME, "Shard has no estimated write load. Decider takes no action.");
+        }
+
+        var allNodeUsageStats = allocation.clusterInfo().getNodeUsageStatsForThreadPools();
+        var nodeUsageStatsForThreadPools = allNodeUsageStats.get(node.nodeId());
+        if (nodeUsageStatsForThreadPools == null) {
+            // No node-level thread pool usage stats were reported for this node. Let's assume this is OK and that the simulator will handle
+            // setting a node-level write load for this node after this shard is assigned.
+            return Decision.single(Decision.Type.YES, NAME, "The node has no write load estimate. Decider takes no action.");
+        }
+
+        assert nodeUsageStatsForThreadPools.threadPoolUsageStatsMap().isEmpty() == false;
+        assert nodeUsageStatsForThreadPools.threadPoolUsageStatsMap().get(ThreadPool.Names.WRITE) != null;
+        var nodeWriteThreadPoolStats = nodeUsageStatsForThreadPools.threadPoolUsageStatsMap().get(ThreadPool.Names.WRITE);
+        var nodeWriteThreadPoolLoadThreshold = writeLoadConstraintSettings.getWriteThreadPoolHighUtilizationThresholdSetting();
+        if (nodeWriteThreadPoolStats.averageThreadPoolUtilization() >= nodeWriteThreadPoolLoadThreshold) {
+            // The node's write thread pool usage stats already show high utilization above the threshold for accepting new shards.
+            String explain = Strings.format(
+                "Node [%s] with write thread pool utilization [%f] already exceeds the high utilization threshold of [%f]. Cannot allocate "
+                    + "shard [%s] to node without risking increased write latencies.",
+                node.nodeId(),
+                nodeWriteThreadPoolStats.averageThreadPoolUtilization(),
+                nodeWriteThreadPoolLoadThreshold,
+                shardRouting.shardId()
+            );
+            logger.debug(explain);
+            return Decision.single(Decision.Type.NO, NAME, explain);
+        }
+
+        if (calculateShardMovementChange(nodeWriteThreadPoolStats, shardWriteLoad) >= nodeWriteThreadPoolLoadThreshold) {
+            // The node's write thread pool usage would be raised above the high utilization threshold with assignment of the new shard.
+            // This could lead to a hot spot on this node and is undesirable.
+            String explain = Strings.format(
+                "The high utilization threshold of [%f] would be exceeded on node [%s] with utilization [%f] if shard [%s] with estimated "
+                    + "write load [%f] (execution time [%f] / threads [%d]) were assigned to it. Cannot allocate shard to node without "
+                    + "risking increased write latencies.",
```

Comment:
I think 
`write load [%f] (execution time [%f] / threads [%d])`
should read
`additional utilisation [%f] (write load [%f] / threads [%d])`
shouldn't it?